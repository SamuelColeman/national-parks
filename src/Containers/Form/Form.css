.form {
	background-image: url('../../images/np_background.jpg');
	margin: 0;
	height: 100vh;
	background-position: center;
	background-repeat: no-repeat;
	background-size: cover;
}

.form_title {
	margin: 0;
  font-size: 15vh;
  padding: 4%;
  color: tan;
}

.form_text {
	font-size: 5vh;
  color: tan;
  margin: 0;
  padding: 3% 0;
}

.form_input {
	border: 5px solid tan;
	background: rgba(0, 0, 0, 0.5);
  width: 7vh;
  height: 6vh;
  font-size: 35px;
  color: tan;
}

.form_button {
	width: 9vh;
  height: 3vh;
  font-size: 2vh;
  background: tan;
  border: 1px solid tan;
  border-radius: 5px;
}

.form_button:hover {
	background: #585F29;
	border: 1px solid #585F29;
}

.form_error {
	color: red;
  font-size: 5vh;
}