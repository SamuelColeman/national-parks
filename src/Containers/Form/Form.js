import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import './Form.css';
import { hasError, getParks, selectState, isLoading } from '../../actions';
import { Link } from 'react-router-dom';

export class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      ConditonalLink: 'div'
    }
  }

  handleChange = (e) => {
    const { selectState, hasError } = this.props;
    selectState(e.target.value.toUpperCase());
    if (e.target.value.length === 2) {
      this.setState({ ConditonalLink: Link });
      hasError('');
    } else {
      hasError('Invalid State');
    }
  }

  render() {
    const { selectedState, errorMsg, submitState } = this.props;
    const { ConditonalLink } = this.state;
    return (
      <section className='form'>
        <h1 className='form_title'>National Parks Directory</h1>
        <h2 className='form_text'>Select State:</h2>
        <input 
          className='form_input'
          type='text' 
          value={selectedState} 
          maxLength='2'
          onChange={this.handleChange} 
          />
          <h2>{errorMsg}</h2>
        <ConditonalLink to='/parks'>
          <button onClick={() => submitState(selectedState)}>Submit</button>
        </ConditonalLink>
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  parks: state.parks,
  selectedState: state.selectedState,
  errorMsg: state.errorMsg
});

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({
    hasError,
    getParks,
    selectState,
    isLoading
  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(Form);