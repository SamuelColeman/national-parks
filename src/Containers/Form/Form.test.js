import React from 'react';
import { shallow } from 'enzyme';
import { Form, mapStateToProps } from './Form';

describe('Form', () => {
	
	let wrapper;
	let mockParks = [{ 
    	description: "Historic Site",
			designation: "National Historic Site",
			directionsInfo: "Drive Car",
			directionsUrl: "http://www.directions.com",
			fullName: "National Historic Site",
			id: "2",
			latLong: "lat:36, long:82",
			name: "Historic Park",
			parkCode: "pant",
			states: "TN",
			url: "https://www.example.com",
			weatherInfo: "Sunny"
	}];
	let mockSelectedState = 'ZN';
	let mockErrorMsg = 'Invalid State';

	beforeEach(() => {
    wrapper = shallow(<Form />)
 	})

  it('should match snapshot with correct data passing through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('map state to props gives the parks array in state', () => {
		const mockState = { parks: mockParks};

		const expected = {
			parks: mockState.parks
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});

	it('map state to props gives the selected state in state', () => {
		const mockState = { selectedState: mockSelectedState};

		const expected = {
			selectedState: mockState.selectedState
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});

	it('map state to props gives the error message in state', () => {
		const mockState = { errorMsg: mockErrorMsg};

		const expected = {
			errorMsg: mockState.errorMsg
		};

		const mappedState = mapStateToProps(mockState);

		expect(mappedState).toEqual(expected);
	});
});