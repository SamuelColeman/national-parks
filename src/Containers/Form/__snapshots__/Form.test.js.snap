// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot with correct data passing through 1`] = `
<section
  className="form"
>
  <h1
    className="form_title"
  >
    National Parks Directory
  </h1>
  <h2
    className="form_text"
  >
    Select State:
  </h2>
  <input
    className="form_input"
    maxLength="2"
    onChange={[Function]}
    type="text"
  />
  <h2 />
  <div
    to="/parks"
  >
    <button
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
</section>
`;
